tests/utils.py

```diff
@@ -0,0 +1,273 @@
+import http
+import json
+import logging
+import os
+import subprocess
+from time import time, sleep
+
+import kubernetes
+import requests
+from ocp_resources.pod import Pod
+from ocp_resources.route import Route
+
+from utilities.constants import (
+    MM_PAYLOAD_PROCESSORS,
+    INFERENCE_ENDPOINT,
+    TRUSTYAI_SERVICE,
+    TRUSTYAI_MODEL_METADATA_ENDPOINT,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class TrustyAIPodNotFoundError(Exception):
+    pass
+
+
+class ModelInputData:
+    def __init__(self, name, num_features, num_observations, data_type):
+        self.name = name
+        self.num_features = num_features
+        self.num_observations = num_observations
+        self.data_type = data_type
+
+
+class TrustyAIModelMetadata:
+    def __init__(
+        self,
+        input_tensor_name,
+        output_tensor_name,
+        num_observations,
+        model_name,
+        num_features,
+    ):
+        self.input_tensor_name = input_tensor_name
+        self.output_tensor_name = output_tensor_name
+        self.num_observations = num_observations
+        self.model_name = model_name
+        self.num_features = num_features
+
+
+def get_ocp_token():
+    token = subprocess.check_output(["oc", "whoami", "-t"]).decode().strip()
+    return token
+
+
+def get_trustyai_pod(client, namespace):
+    pod = next(
+        (
+            pod
+            for pod in Pod.get(dyn_client=client, namespace=namespace.name)
+            if TRUSTYAI_SERVICE in pod.name
+        ),
+        None,
+    )
+    if pod is None:
+        raise TrustyAIPodNotFoundError(
+            f"No TrustyAI pod found in namespace {namespace.name}"
+        )
+    return pod
+
+
+def get_trustyai_service_route(namespace):
+    try:
+        return Route(namespace=namespace.name, name=TRUSTYAI_SERVICE)
```

Comment:
this will never raise anything Route(namespace=namespace.name, name=TRUSTYAI_SERVICE)

you need to do 
```
route = Route(namespace=namespace.name, name=TRUSTYAI_SERVICE)
if not route.exists:
    raise
......


```

Reply:
done